### ISSUE TITLE
strih OBS crash c0000005 v distroav ndi_source_update -> obs new_prop (NULL-prop pri 4.7s render stalle pocas E2E [4c] enforce)

### ISSUE THREAD

**Issue by zbynekdrlik** [zbynekdrlik/camera-box#1224]
**Signatúra (strih, Crash 2026-08-30 04-13-20.txt, libobs 32.2.0):**
```
Unhandled exception: c0000005
Fault address: obs.dll!new_prop+0xa2
Crashed thread:
  obs.dll!new_prop+0xa2
  distroav.dll!ndi_source_update+0x1e8
  distroav.dll!new_ndi_receiver_name+0xa2   (symbolizácia offsetová — najbližší export)
```

**Kontext:** 04:13:19 lokálne = 02:13:19 UTC = presne [4c/8]/pre-record fáza E2E behu 225200112 (nahrávanie začalo ~02:13:43 — po crashi AHK respawol OBS o 04:13:57 a beh prežil, verdikt sa vypočítal). Pred pádom OBS log ukazuje ťažký stall: `genlock #797 slow output_video 'NDI 2ME PVW': total=4685.27ms` + `Audio timestamp for 'ASIO zvuk' exceeded TS_SMOOTHING_THRESHOLD, diff=4645469674 ns` + viac vstupov s empty_run — čiže `ndi_source_update` (naše WS zápisy ndi settings počas enforce/heal) bežal proti wedgnutému grafu a v properties ceste (`new_prop`) narazil na NULL.

**Trieda:** vendored DistroAV update path NULL-deref — príbuzné vendored-obs-frontend-crash-safety doktríne (guard at the CONSUMER; obs_data_get_json NULL trieda z issue 773). Fix smer: v `ndi-source.cpp` update/properties ceste guardovať NULL návraty pred `new_prop`/obs_properties skladaním, prípadne zamknúť update proti concurrent property build. Vendored zmena = plný lock-step anchors režim (tests/genlock_preload.rs + oba windows-genlock yml), CI-first compile.

**Výskyt:** prvý dump tejto signatúry (crashes dir: predtým 08-06, 08-02, 07-30 — iné). Zatiaľ 1×; ak sa zopakuje počas [4c] enforce, priorita hore.

Scope-gate: planned-work

**Comment by zbynekdrlik** [comment]
## STEP-0 validácia (worker lane-1224, proti aktuálnemu `dev` @ d69f6b3cb)

Preveril som crash signatúru proti CURRENT CODE a **potvrdzujem, že cesta je STÁLE VALID** — defektná (neguardovaná) consumer cesta k `obs.dll!new_prop` v `vendor/distroav/src/ndi-source.cpp` existuje nezmenená v `dev`.

Čo som overil (čítaním kódu, nie zo stale ticket textu):

1. **`new_prop` je v distroav dosiahnuteľný IBA cez `ndi_source_getproperties`.** `new_prop` (`vendor/obs-studio/libobs/obs-properties.c:436`) volajú len `obs_properties_add_*`. V distroav ich volá jedine `ndi_source_getproperties` (add_list/add_bool×3/add_int na riadkoch 618/638/646/653/659). **`ndi_source_update` NEMÁ žiadnu cestu k `new_prop`** — celé telo (riadky 1670–1968) číta/zapisuje settings + spúšťa receiver thread, nikdy neskladá `obs_properties`. Preto sú crash rámce `ndi_source_update+0x1e8`/`new_ndi_receiver_name+0xa2` offsetovo mis-symbolizované (ticket to sám značí „najbližší export") a reálne padajúca distroav funkcia je `ndi_source_getproperties`.

2. **Neguardovaná consumer cesta reálne existuje v aktuálnom kóde:**
   - `ndi_source_getproperties` (r.610) `obs_properties_t *props = obs_properties_create();` — **žiadny NULL guard** pred `obs_properties_add_*` skladaním.
   - `obs_property_t *source_list = obs_properties_add_list(...)` (r.618) — **žiadny NULL guard** pred tým, ako sa `source_list` odovzdá do finder lambda + `genlock_ensure_saved_source_listed`.
   - `genlock_ensure_saved_source_listed` (r.573) guarduje `!s` ale **NEGUARDUJE `!source_list`**.

3. **Async lifetime hazard confirmed live v kóde:** `finder.getNDISourceList(finder_callback)` (`vendor/distroav/src/ndi-finder.cpp:16`) spúšťa callback na **DETACHED threade** (`std::thread(refreshNDISourceList, callback).detach()`), ktorý fajruje 5+ s neskôr (5 s throttle + `find_wait_for_sources(…,1000)` slučka), teda dávno po tom, čo `ndi_source_getproperties` vrátil `props`. Lambda zachytáva RAW `source_list`/`s` a potom volá `obs_source_update_properties(s->obs_source)` (neguardované `s`), čo spustí ČERSTVÝ `ndi_source_getproperties` → `obs_properties_add_*` → `new_prop`. To presne sedí na crash kontext (WS enforce/heal `SetInputSettings` churn počas 4.7 s render stallu → wedged graf).

Behaviorálny repro (živý NULL/UAF pod render stallom) NIE JE lokálne reprodukovateľný — vendored C/C++ sa kompiluje len na CI a crash sa prejaví len na živom rigu; validácia je statická (čítanie aktuálneho kódu + potvrdenie, že neguardovaná cesta je prítomná). Ticket ostáva VALID → pokračujem defenzívnym guardom pri consumerovi.

**Comment by zbynekdrlik** [comment]
## Dizajn — #1224 strih OBS c0000005 v distroav properties ceste → `obs.dll!new_prop` (NULL/lifetime pod render stallom)

Triage: **non-trivial** — vendored (DistroAV) crash-safety s concurrency/lifetime rozmerom, viac obhájiteľných prístupov s rôznymi dôsledkami, CI-first kompilácia, žiadny lokálny behaviorálny repro.

### Root cause (vytrasovaný v kóde, nie symptóm)

Crash je c0000005 v `obs.dll!new_prop+0xa2` — presne v `HASH_ADD_STR(props->properties, name, p)` časti `new_prop` (`vendor/obs-studio/libobs/obs-properties.c:436`), t.j. NULL/zlý pointer pri skladaní `obs_property`. `new_prop` je z distroav dosiahnuteľný **JEDINE cez `ndi_source_getproperties`** (volá `obs_properties_add_list/bool/int`). `ndi_source_update` NEMÁ žiadnu cestu k `new_prop` — crash rámce `ndi_source_update+0x1e8`/`new_ndi_receiver_name+0xa2` sú offsetovo mis-symbolizované (ticket to sám značí „najbližší export"), reálne padá `ndi_source_getproperties`.

Konkrétny lifetime hazard: `finder.getNDISourceList(finder_callback)` (`ndi-finder.cpp:16`) spúšťa `finder_callback` na **DETACHED threade** (`std::thread(...).detach()`), ktorý fajruje 5+ s neskôr (5 s throttle + `find_wait_for_sources(…,1000)` slučka) — dávno po tom, čo `ndi_source_getproperties` vrátil `props`. Lambda zachytáva RAW `source_list` (vlastnený tým `props`) a `s` (`ndi_source_t*`), potom volá `obs_source_update_properties(s->obs_source)`, čo spustí ČERSTVÝ `ndi_source_getproperties` → `obs_properties_add_*` → `new_prop`. Počas WS enforce/heal `SetInputSettings` churnu na 4.7 s render-stallnutom grafe (kontext z OBS logu) je to use-after-free / NULL-deref na neguardovaných consumeroch. Ďalšie neguardované miesta na tej istej ceste: `obs_properties_create()` (NULL pri OOM pod pamäťovým tlakom stallu), `obs_properties_add_list` návrat, a `genlock_ensure_saved_source_listed` guarduje `!s` ale nie `!source_list`.

### Prístup 1 — defenzívne NULL-guardy PRI CONSUMEROVI + `[distroav]` warn (ZVOLENÝ)

Presne doktrína `vendored-obs-frontend-crash-safety.md`: guard at the CONSUMER, early-return, loud warn. Aditívne, bez zmeny happy-path správania:
- `ndi_source_getproperties`: guard `!props` (early-return `nullptr`) a `!source_list` (preskočiť len source-list-závislý blok — finder + inject — a vrátiť ostatné whitelist properties).
- finder callback: guard `!source_list || !s || !s->obs_source` na začiatku — zastaví aj spurious `obs_source_update_properties` refresh nad mŕtvym `s`, čo je práve trigger reštartu getproperties→new_prop cesty.
- `genlock_ensure_saved_source_listed`: pridať `!source_list` do existujúceho guardu.

Trade-off / výhody: minimálny, revert-safe, žiadny deadlock risk; nechytí non-NULL freed UAF (freed-but-non-NULL pointer) — to je hlbší lifetime problém, honestly priznaný a eskalovateľný ak sa crash zopakuje (ticket: „zatiaľ 1×; ak sa zopakuje, priorita hore").

### Prístup 2 — zamknúť properties build proti `ndi_source_update` (config_mutex) — ZAMIETNUTÝ

Ticketom navrhnutá alternatíva. Zamietnutý: `#93` komentár v `ndi_source_update` výslovne varuje pred deadlockom medzi `config_mutex` a `pthread_join` v `ndi_source_thread_stop`; `ndi_source_getproperties` beží na UI/Qt threade a rozšírenie zámku naň je redesign s reálnym deadlock rizikom. Navyše NErieši detached-thread UAF (callback beží mimo akéhokoľvek update). Ťažší a rizikovejší než guard, mimo „defensive guard, not redesign" mandátu.

### Prístup 3 — redizajn async finder callbacku (weak-ref/refcount `s`, nezachytávať raw `source_list`) — ZAMIETNUTÝ

Najúplnejšie by odstránil UAF, ale je to redesign upstream DistroAV štruktúry (mandát: 1× výskyt, defenzívny guard nie redesign). Zaznamenané ako follow-up, ktorý sa otvorí len ak crash recidivuje.

### Architektúra

Štruktúra crash-reaching cesty a kam guard patrí (žiadny nový modul/framework — použije sa EXISTUJÚCA DistroAV topológia + kanonický guard-at-consumer vzor):

- Topológia volania: `ndi_source_getproperties` (sync, UI thread) → `obs_properties_create` → `obs_properties_add_list` → `new_prop`; paralelne detached finder thread → `finder_callback` → `obs_source_update_properties(s->obs_source)` → (znova) `ndi_source_getproperties` → `new_prop`. Guardy sa umiestnia presne na tieto existujúce consumer uzly, nie do libobs.
- Framework/vzor: `vendored-obs-frontend-crash-safety.md` (#773 trieda `obs_data_get_json`-NULL → guard-at-consumer) je priamo aplikovateľný vzor; nič nové sa nevymýšľa. Zmena je čisto aditívna (early-return + warn), takže topológia a všetky existujúce statické anchory (`genlock_ensure_saved_source_listed(source_list, s)` call-site, ktorý anchorujú OBA `windows-genlock*.yml`) ostávajú byte-identical — call-sites nemeníme, guardy idú do tiel funkcií/lambdy.
- Testovacia štruktúra: `#773`-trieda NULL-guard = Rust source-anchor test IBA (`tests/distroav_getproperties_null_guard_1224.rs`, std-only, #1026 rustc recipe), BEZ pwsh mirroru (crash-safety pravidlo: pure NULL/crash-safety guard → Rust anchor only; behaviorálna divergencia by dostala pwsh mirror, toto nie je ona). Vendored C++ sa kompiluje len na CI; lokálne nesie zmenu hand-audit + `cargo fmt --all --check` (parsuje Rust test) + anchor RED→GREEN cez rustc.

**Comment by zbynekdrlik** [comment]
## Review pass (CYCLE step 6) — worker lane-1224

Spustený adversarial **review** (kombinované `/review` + `/requesting-code-review`) cez gated-Fable dispatch (fable-gate OPEN, `model: fable`, xhigh) proti diffu `d69f6b3cb..HEAD` (vendored `ndi-source.cpp` guardy + std-only anchor test + version bump). Reviewer verifikoval tvrdenia proti reálnym vendored zdrojom, nie len proti diff textu.

**Verdikt (pred opravou): 0 🔴 1 🟡 2 🔵 → po oprave (commit f6890aa01): 0 🔴 0 🟡 0 🔵 (clean).**

Čo review potvrdil ako čisté: happy-path správanie nezmenené; `return nullptr` je platný návrat pre `ndi_source_getproperties` (`obs_properties_apply_settings` guarduje `!props`); finder-callback early-return predchádza VŠETKÝM derefom vrátane `obs_source_update_properties`; #414 štrukturálne bez porušenia (čisto aditívne guardy + 1 std-only test); anchor test je reálny RED→GREEN (base má 0 výskytov `!source_list || !s || !s->obs_source`, po oprave 2); oba `windows-genlock*.yml` anchory (`genlock_ensure_saved_source_listed(source_list, s)`) ostali byte-identical; Rust-anchor-only / no-pwsh-mirror je správna trieda pre pure crash-safety guard.

Nálezy — všetky opravené v **f6890aa01** (iba komentáre/doc, žiadna zmena logiky, per #311 same-branch mandate + HARD no-overstatement pravidlo):

- 🟡 **W1 — komentáre precenili pokrytie.** `!props` guard je dokázateľne mŕtvy kód na stock libobs: `obs_properties_create`→`bmalloc`, ktorý pri OOM volá `bcrash()` (nikdy nevráti NULL), a každý `obs_properties_add_*` early-returuje na `!props` PRED `new_prop`. `new_prop+0xa2` crash je non-NULL corruption/UAF, ktorý žiadny NULL guard nechytí. Genuinely-closed sub-class je detached finder-callbackov `s->obs_source` NULL deref (libobs volá `get_properties(data=NULL)` pre type-level buildy, takže `s==NULL` je reálny živý prípad; pre-fix callback potom bežal `obs_source_update_properties(s->obs_source)` = NULL deref na finder threade v distroav.dll, konzistentné s dump rámcami). Komentáre + tvrdenie o pokrytí prepísané na čestné: fix zatvára `s==NULL` detached-callback deref; presná `new_prop` UAF signatúra NIE JE dokázateľne odstránená (accepted residual — 1× výskyt, bez lockingu kvôli `config_mutex`/`pthread_join` deadlock riziku).
- 🔵 **S1** — `!source_list` v `genlock_ensure_saved_source_listed` je redundantné (libobs list ops sú NULL-`p`-safe); ponechané ako explicitné belt-and-braces, komentár už mu nepripisuje crash-prevention.
- 🔵 **S2** — test doc: lokálny rustc recept potrebuje `CARGO_MANIFEST_DIR=$PWD` (compile-time `env!`), doplnené.

Poznámka k pokrytiu (no-overstatement): tento fix je DEFENZÍVNY guard, ktorý spoľahlivo zatvára NULL sub-triedu crash cesty; presnú `new_prop` UAF signatúru nedokáže lokálne reprodukovať ani garantovať odstránenie (vendored kód sa kompiluje len na CI, crash sa prejaví len na živom rigu). Ak sa signatúra zopakuje počas `[4c]` enforce, priorita hore + hlbší lifetime redesign (weak-ref/refcount async closure) — ako ticket sám hovorí.

**Comment by zbynekdrlik** [comment]
## Deploy note (supervisor) — worker lane-1224

Táto oprava mení VENDORED plugin kód (`vendor/distroav/src/ndi-source.cpp` → `distroav.dll` na strih/stream, `distroav.so` na imag). Po merge NIE JE živá, kým sa nenasadí:

- **FULL-BUNDLE deploy** (nie fast-dll obs.dll swap) na strih + stream + imag — per `.claude/rules/rig-state-inspection.md` (bod 1 a 5): merged `vendor/<plugin>/**` fix je live až po fleet deploy; distroav.dll na Windows žije v `C:\ProgramData\obs-studio\plugins\distroav\bin\64bit\`, imag musí dostať CELÝ matched bundle (nie len jeden `.so`).
- Deployment + post-deploy overenie je SUPERVISOROV krok (worktree lane, full authority) — worker sa zastavuje na zelenom lokálnom review.
- CI je prvá kompilácia tohto C++ (Tier-0: žiadny lokálny plugin build); `linux-genlock.yml` (imag parity) + `windows-genlock.yml` sú relevantné buildy.

**Comment by zbynekdrlik** [comment]
Fix is LIVE fleet-wide — closing.

- NULL-guards merged (vendored distroav ndi-source.cpp consumer guards; CI test tests/distroav_getproperties_null_guard_1224.rs green in run 33321437692 on dev 3eb21b2ed).
- Full-bundle fleet deploy 2026-08-31 from ONE anchor (windows run 33350586986 + linux run 33351782460, same SHA 3eb21b2ed via deploy-genlock-fleet.sh):
  - strih: obs.dll sha b072ad12… + ProgramData distroav.dll sha e788c337… byte-verified; relaunched session 1, titlebar "build 3eb21b2ed", render tick ENABLED, distroav 6.2.1 loaded, AHK count 1.
  - stream: same DLL shas byte-verified; launch verify PASS (session 1, titlebar build 3eb21b2ed, render tick ENABLED, DistroAV loaded); bundle-state reports genlock_build_sha=3eb21b2ed.
  - imag: whole-bundle install + imag-obs.service SUPERVISED restart (pid in unit cgroup, "OK: OBS bezi"), GENLOCK_BUILD_SHA=3eb21b2ed.
- Rig re-pinned to TEST mode after the deploy (rig-mode.sh test: burns ON, chain verified end to end).

The crash's [4c/8] repro window returns with the E2E runs post-recable; the guard class is regression-locked in CI either way.
